src/memkind_pmem.c

```diff
@@ -100,9 +100,12 @@ bool pmem_extent_dalloc(extent_hooks_t *extent_hooks,
     // if madvise fail, it means that addr isn't mapped shared (doesn't come from pmem)
     // and it should be unmapped to avoid space exhaustion when calling large number of
     // operations like memkind_create_pmem and memkind_destroy_kind
-    // EOPNOTSUPP is returned in case of filesystem doesn't support FALLOC_FL_PUNCH_HOLE
     errno = 0;
-    if (madvise(addr, size, MADV_REMOVE) != 0 && errno != EOPNOTSUPP) {
+    if (madvise(addr, size, MADV_REMOVE) != 0) {
+        if (errno == EOPNOTSUPP) {
+            log_fatal("Filesystem doesn't support FALLOC_FL_PUNCH_HOLE flag");
```

Comment:
My suggestion is that it is better to log:
"Filesystem doesn't support FALLOC_FL_PUNCH_HOLE operation" than "Filesystem doesn't support FALLOC_FL_PUNCH_HOLE flag".